drivers/dai/intel/ssp/dai-params-intel-ipc4.h

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __DAI_PARAMS_INTEL_IPC4_H__
+#define __DAI_PARAMS_INTEL_IPC4_H__
+
+#include <stdint.h>
+
+#define I2S_TDM_INVALID_SLOT_MAP1 0xF
+#define I2S_TDM_MAX_CHANNEL_COUNT 8
+#define I2S_TDM_MAX_SLOT_MAP_COUNT 8
+
+/**< Type of the gateway. */
+enum ipc4_connector_node_id_type {
+	/**< HD/A host output (-> DSP). */
+	ipc4_hda_host_output_class = 0,
+	/**< HD/A host input (<- DSP). */
+	ipc4_hda_host_input_class = 1,
+	/**< HD/A host input/output (rsvd for future use). */
+	ipc4_hda_host_inout_class = 2,
+
+	/**< HD/A link output (DSP ->). */
+	ipc4_hda_link_output_class = 8,
+	/**< HD/A link input (DSP <-). */
+	ipc4_hda_link_input_class = 9,
+	/**< HD/A link input/output (rsvd for future use). */
+	ipc4_hda_link_inout_class = 10,
+
+	/**< DMIC link input (DSP <-). */
+	ipc4_dmic_link_input_class = 11,
+
+	/**< I2S link output (DSP ->). */
+	ipc4_i2s_link_output_class = 12,
+	/**< I2S link input (DSP <-). */
+	ipc4_i2s_link_input_class = 13,
+
+	/**< ALH link output, legacy for SNDW (DSP ->). */
+	ipc4_alh_link_output_class = 16,
+	/**< ALH link input, legacy for SNDW (DSP <-). */
+	ipc4_alh_link_input_class = 17,
+
+	/**< SNDW link output (DSP ->). */
+	ipc4_alh_snd_wire_stream_link_output_class = 16,
+	/**< SNDW link input (DSP <-). */
+	ipc4_alh_snd_wire_stream_link_input_class = 17,
+
+	/**< UAOL link output (DSP ->). */
+	ipc4_alh_uaol_stream_link_output_class = 18,
+	/**< UAOL link input (DSP <-). */
+	ipc4_alh_uaol_stream_link_input_class = 19,
+
+	/**< IPC output (DSP ->). */
+	ipc4_ipc_output_class = 20,
+	/**< IPC input (DSP <-). */
+	ipc4_ipc_input_class = 21,
+
+	/**< I2S Multi gtw output (DSP ->). */
+	ipc4_i2s_multi_link_output_class = 22,
+	/**< I2S Multi gtw input (DSP <-). */
+	ipc4_i2s_multi_link_input_class = 23,
+	/**< GPIO */
+	ipc4_gpio_class = 24,
+	/**< SPI */
+	ipc4_spi_output_class = 25,
+	ipc4_spi_input_class = 26,
+	ipc4_max_connector_node_id_type
+};
+
+/**< Invalid raw node id (to indicate uninitialized node id). */
+#define IPC4_INVALID_NODE_ID	0xffffffff
```

Comment:
ditto prefix changes for all structs, enum and macros

Reply:
ok